File: flutter_app/lib/models/branch.dart
class Branch {
  final int id;
  final String name;
  final String description;
  final String address;
  final String city;
  final String state;
  final String zipCode;
  final double? latitude;
  final double? longitude;
  final String phone;
  final String email;
  final String openingTime;
  final String closingTime;
  final bool isActive;
  final List<String> images;
  final String createdAt;
  final double? distanceKm;

  const Branch({
    required this.id,
    required this.name,
    this.description = '',
    this.address = '',
    this.city = '',
    this.state = '',
    this.zipCode = '',
    this.latitude,
    this.longitude,
    this.phone = '',
    this.email = '',
    this.openingTime = '08:00',
    this.closingTime = '22:00',
    this.isActive = true,
    this.images = const [],
    this.createdAt = '',
    this.distanceKm,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      address: json['address'] as String? ?? '',
      city: json['city'] as String? ?? '',
      state: json['state'] as String? ?? '',
      zipCode: json['zip_code'] as String? ?? '',
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      phone: json['phone'] as String? ?? '',
      email: json['email'] as String? ?? '',
      openingTime: json['opening_time'] as String? ?? '08:00',
      closingTime: json['closing_time'] as String? ?? '22:00',
      isActive: json['is_active'] as bool? ?? true,
      images: (json['images'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
      createdAt: json['created_at'] as String? ?? '',
      distanceKm: (json['distance_km'] as num?)?.toDouble(),
    );
  }

  String get fullAddress {
    final parts = [address, city, state].where((s) => s.isNotEmpty).toList();
    return parts.join(', ');
  }

  String get distanceText {
    if (distanceKm == null) return '';
    if (distanceKm! < 1) return '${(distanceKm! * 1000).round()} m';
    return '${distanceKm!.toStringAsFixed(1)} km';
  }
}

File: app/models/branch.py
from ..database.db import db
from ..utils import now_mx
import json


class Branch(db.Model):
    __tablename__ = "branches"
    id           = db.Column(db.Integer, primary_key=True)
    name         = db.Column(db.String(120), nullable=False)
    description  = db.Column(db.Text, default="")
    address      = db.Column(db.String(255), default="")
    city         = db.Column(db.String(100), default="")
    state        = db.Column(db.String(100), default="")
    zip_code     = db.Column(db.String(20), default="")
    latitude     = db.Column(db.Float, nullable=True)
    longitude    = db.Column(db.Float, nullable=True)
    phone        = db.Column(db.String(30), default="")
    email        = db.Column(db.String(255), default="")
    opening_time = db.Column(db.String(10), default="08:00")
    closing_time = db.Column(db.String(10), default="22:00")
    is_active    = db.Column(db.Boolean, default=True)
    images       = db.Column(db.Text, default="[]")  # JSON array
    created_at   = db.Column(db.DateTime, default=now_mx)

    inventory   = db.relationship("BranchInventory", backref="branch", lazy=True, cascade="all, delete-orphan")
    reviews     = db.relationship("BranchReview", backref="branch", lazy=True, cascade="all, delete-orphan")
    assignments = db.relationship("BranchAssignment", backref="branch", lazy=True, cascade="all, delete-orphan")
    sales       = db.relationship("Sale", backref="branch", lazy=True)

    def to_dict(self, distance_km=None):
        try:
            images = json.loads(self.images) if self.images else []
        except Exception:
            images = []
        d = {
            "id": self.id,
            "name": self.name,
            "description": self.description or "",
            "address": self.address or "",
            "city": self.city or "",
            "state": self.state or "",
            "zip_code": self.zip_code or "",
            "latitude": self.latitude,
            "longitude": self.longitude,
            "phone": self.phone or "",
            "email": self.email or "",
            "opening_time": self.opening_time or "08:00",
            "closing_time": self.closing_time or "22:00",
            "is_active": self.is_active,
            "images": images,
            "created_at": self.created_at.isoformat(),
        }
        if distance_km is not None:
            d["distance_km"] = distance_km
        return d
